logic/lblockindex/lblockindex_test.go

```diff
@@ -0,0 +1,466 @@
+package lblockindex
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/copernet/copernicus/conf"
+	"github.com/copernet/copernicus/log"
+	"github.com/copernet/copernicus/model"
+	"github.com/copernet/copernicus/model/block"
+	"github.com/copernet/copernicus/model/blockindex"
+	"github.com/copernet/copernicus/model/chain"
+	"github.com/copernet/copernicus/model/pow"
+	"github.com/copernet/copernicus/model/utxo"
+	"github.com/copernet/copernicus/persist"
+	"github.com/copernet/copernicus/persist/blkdb"
+	"github.com/copernet/copernicus/persist/db"
+	"github.com/copernet/copernicus/persist/disk"
+	"github.com/copernet/copernicus/util"
+	"io/ioutil"
+	"math/big"
+	"math/rand"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+)
+
+var pathblockdb string
+var pathutxodb string
+
+func initBlockDB() {
+	var err error
+	pathblockdb, err = ioutil.TempDir("/tmp", "blockindextest")
+	if err != nil {
+		panic(fmt.Sprintf("generate temp db path failed: %s\n", err))
+	}
+	bc := &blkdb.BlockTreeDBConfig{
+		Do: &db.DBOption{
+			FilePath:  pathblockdb,
+			CacheSize: 1 << 20,
+		},
+	}
+
+	blkdb.InitBlockTreeDB(bc)
+}
+
+func initUtxoDB() {
+	var err error
+	pathutxodb, err = ioutil.TempDir("/tmp", "utxotest")
+	if err != nil {
+		panic(fmt.Sprintf("generate temp db path failed: %s\n", err))
+	}
+
+	dbo := db.DBOption{
+		FilePath:       pathutxodb,
+		CacheSize:      1 << 20,
+		Wipe:           false,
+		DontObfuscate:  false,
+		ForceCompactdb: false,
+	}
+
+	uc := &utxo.UtxoConfig{
+		Do: &dbo,
+	}
+	utxo.InitUtxoLruTip(uc)
+}
+
+func initEnv() {
+	model.SetRegTestParams()
+	conf.Cfg = &conf.Configuration{}
+	conf.Cfg.DataDir = "/tmp"
```

Comment:
you can use
```
args := []string{"--datadir=/tmp/xxx"}
conf.Cfg = conf.InitConfig(args)
```
replace 
```
conf.Cfg = &conf.Configuration{}
conf.Cfg.DataDir = "/tmp"
```

Reply:
Yes, your code is more formal. But InitConfig() will find conf file on disk, and do some other logic. Here, this unit test only need to use one config. So I write for convenience.